kymatio/scattering2d/frontend/torch_frontend.py

```diff
@@ -13,6 +13,8 @@ def __init__(self, J, shape, L=8, max_order=2, pre_pad=False,
         ScatteringBase2D._instantiate_backend(self, 'kymatio.scattering2d.backend.')
         ScatteringBase2D.build(self)
         ScatteringBase2D.create_filters(self)
+        if pre_pad:
+            self.pad = lambda x: x.reshape(x.shape + (1,))
```

Comment:
Did we discuss this? This seems awkward since the padding seems to be casting to our real data type, so once again the padding is not doing what it was advertised to do.

Reply:
I guess this is also about backwards compatibility?

Reply:
I think this is fine.